### ISSUE TITLE
`indent` broken with object parameter on new line when using Allman brace style

### ISSUE THREAD

**Issue by swernerx** [eslint/eslint#5548]
**What version of ESLint are you using?**
v2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Espree

**What did you do? Please include the actual source code causing the issue.**
```js
import { formatNumber, parseToNumber } from "utils"

export default function install(Vue)
{
  Vue.filter("format-percent",
  {
    read: function(val)
    {
      return formatNumber(val)
    },

    write: function(val)
    {
      return parseToNumber(val)
    }
  })
}
```

My configuration:

```yml
env:
  browser: true
  es6: true
  node: true
  worker: true
  serviceworker: true

parserOptions:
  ecmaVersion: 6
  sourceType: module

rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
      VariableDeclarator: 0

  linebreak-style:
    - error
    - unix

  quotes:
    - error
    - double
    - avoid-escape

  semi:
    - error
    - never

  brace-style:
    - error
    - allman
    - allowSingleLine: true
```

Seems to be related to this report: https://github.com/eslint/eslint/issues/5064

**What did you expect to happen?**

That no error is being shown.

**What actually happened? Please include the actual, raw output from ESLint.**

```
⚡ eslint test.js

/Users/Sebastian/eslinttest/test.js
   6:3  error  Expected indentation of 4 space characters but found 2  indent
   7:5  error  Expected indentation of 6 space characters but found 4  indent
  12:5  error  Expected indentation of 6 space characters but found 4  indent
  16:3  error  Expected indentation of 4 space characters but found 2  indent

✖ 4 problems (4 errors, 0 warnings)
```

**Comment by swernerx** [comment]
One part of the issue seems to be the result of the detailed configuration of `allowSingleLine`. If this setting is not defined it reports these issues:

```
⚡ eslint brace-*

/Users/Sebastian/eslinttest/brace-wrong.js
   1:1   error  Opening curly brace appears on the same line as controlling statement  brace-style
  16:14  error  Opening curly brace appears on the same line as controlling statement  brace-style

✖ 2 problems (2 errors, 0 warnings)
```

That's still ignoring array, objects and es6 classes.

**Comment by platinumazure** [comment]
@BYK Should this be labeled "evaluating", not "triage"?

**Comment by ilyavolodin** [comment]
This is a known issue. Right now our indent rule doesn't handle multi-line parameters correctly. I don't think this is related to allman style, I think the same issue will happen with 1tbs as well. @gyandeeps @BYK I think this can be closed as a duplicate, but could you verify please?

**Comment by BYK** [comment]
Actually, I think the rule is right here. That object is part of a function call and the brace should be indented. Right?

**Comment by ilyavolodin** [comment]
Looking at this again, I think @BYK is correct, brace on the 6th line is part of the argument of a function and should be indented an extra level, so does everything following it. Closing as the rule seems to be doing the right thing in this case. If you disagree, feel free to reopen.

**Comment by swernerx** [comment]
I am more in favor that this is not how the rule should behave. Actually when the last parameter is a callback or option block I favor having all of it on the same column and not indented. The reasoning is that in this case it's pretty comparable to a normal function body where we also do not indent the curly braces.

Example with option dict:

```js

test.serial("should accept promised content", (t) =>
  compareFixtures(t, "custom-load",
  {
    load: () =>
      Promise.resolve("custom-content {}")
  })
)
```

from my usage of the ava test runner.

**Comment by swernerx** [comment]
Probably a work around for those loving this kind of allman style would be to have an option for `indent` to disable argument alignment e.g. for `ObjectExpression` and `FunctionExpression`. That's a different use case than when splitting long parameter lists to multiple lines (which is questinable whether the issue is another then). 

As an example, I think that this code style is totally okay:

```js
test.serial("should accept promised content",
{
  foo: "bar"
}, 
function() 
{
  doSomething()
})
```
